lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -530,12 +553,65 @@ isolatePut l x = do
         ++ ", but expected to be "
         ++ (show l)
 
-withSizeHeader16be :: Put -> Put
-withSizeHeader16be x = do
-    let bs = BL.toStrict $ runPut x
+-- | Add a corresponding header to a message. Every message is encoded as:
+--
+--     HEADER(MESSAGE) | MESSAGE
+--
+-- where `HEADER` is:
+--
+--     SIZE (2 bytes) | TYPE (1 byte)
+--
+withHeader :: MessageType -> Put -> Put
+withHeader typ content = do
+    let bs = BL.toStrict $ runPut (putWord8 (messageTypeTag typ) *> content)
     putWord16be (toEnum $ BS.length bs)
     putByteString bs
 
+-- | Compute a Blake2b_256 hash of a given 'ByteString'
+blake2b256 :: ByteString -> ByteString
```

Comment:
maybe use `blake2b256` from `lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Random.hs` ?

Reply:
Hmmm... I'd rather move this function in a common place that is neither Jormungandr.Binary, nor AddressDerivation.Random ^^" ... But good point.